### PR TITLE
Lite interpret

### PR DESCRIPTION
**Pull request by gferraro**
- This fixes a few bugs and adds a class to convert a  model to tflite
- Adds ability to use a neuralstick model, tflite or tf model on the pi
- Fixes bug with cache_to_disk crashing load
- Changes the build process to keep a camera-location combo in only one dataset (train, validate, test)
- Ability to reproccess in load.py, but by default skip clips that are already present in the database
- Added resnet model
- Fixed bug with last filtered frame not being correct
- Use tf2.0

### CODE REVIEW

File: ml_tools/dataset.py [hunk 1]
```diff
@@ -615,21 +660,28 @@ def load_tracks(self):
...
-            if self.add_track_header(track):
+            if self.add_track_header(track, max_segments_per_track):
                 result += 1
         return result
 
-    def add_track_header(self, track_header):
+    def add_track_header(self, track_header, max_segments_per_track=None):
         if track_header.track_id in self.track_by_id:
             return False
+
```
Comment (clare) [inline ml_tools/dataset.py]: Taking random track segments could mean that the segments still overlap I think (as from memory segments can overlap some of the frames) - don't worry about fixing that now.

File: piclassifier/piclassify.py [hunk 2]
```diff
@@ -28,6 +29,93 @@
...
+        stats = json.load(open(filename + ".txt", "r"))
+
+        self.MODEL_NAME = stats["name"]
+        self.MODEL_DESCRIPTION = stats["description"]
+        self.labels = stats["labels"]
+        self.eval_score = stats["score"]
+        self.params = stats["hyperparams"]
+
+
+# TO DO TFLIteInterpreter
```
Comment (clare) [inline piclassifier/piclassify.py]: Is this still needed to be done?

File: ml_tools/dataset.py (hunk 1 above)
Comment (gferraro) [reply ml_tools/dataset.py]: That is true, i wasn't worried so much about that.In the last build I increased the segment spacing to 2 seconds. 

I'm not sure if we need to be worried about having too many segments from a track. if we have enough data. But a whole bunch of work could be doing to improving the distribution.

File: load/cliptrackextractor.py [hunk 3]
```diff
@@ -149,23 +157,34 @@ def _whole_clip_stats(self, clip, frames):
             ):
                 clip.background = None
 
-    def process_frames(self, clip, frames):
+    def process_frames(self, clip, raw_frames):
         # for now just always calculate as we are using the stats...
         # background np.float64[][] filtered calculated here and stats
-        clip.background_from_whole_clip(frames)
-        self._whole_clip_stats(clip, frames)
+        non_ffc_frames = [
```
[the remark this answers is not in the document]
Comment (gferraro) [reply load/cliptrackextractor.py]: we are converting from an int version, but you are correct this does not need to be done. If the array is already float32, it wont create a copy btw.

File: classifier_TEMPLATE.yaml [hunk 4]
```diff
@@ -183,16 +184,29 @@ train:
 
         # model
         batch_norm: true
-        lstm_units: 256
+        lstm_units: 512
```
[the remark this answers is not in the document]
Comment (gferraro) [reply classifier_TEMPLATE.yaml]: Just using the numbers that I used to build the model, and that were used in the model we had been using

**Pull request merged**